### DISCUSSION TITLE
Can I retry when meet  `Pod was active on the node longer than the specified deadline`  error.

### DISCUSSION THREAD

**Discussion by zhengliquan** [argoproj/argo-workflows#8464] [Q&A]
I want to set a strategy to retry when meet `
Pod was active on the node longer than the specified deadline`. But found no results.
I tried:
1. set a environment variable `TRANSIENT_ERROR_PATTERN` and set the value to `.*Pod was active on the node longer than the specified deadline.*`.
2. use `expression`: set the expression to `'{{ workflow.failures }}' contains 'Pod was active on the node longer than the specified deadline'`.

I think the second method should be OK, but I have no idea about how to get the specific workflow failures info or the error code of `Pod was active on the node longer than the specified deadline.` If there is an error code, may try to use `lastRetry.exitCode`.

Thanks for help~

**Comment by terrytangyuan** [comment]
TRANSIENT_ERROR_PATTERN should work. Make sure you included it in the controller and executor deployment's environment variables. 

Also you could increase the deadline in your pod spec.

**Reply by zhengliquan** [reply]
Thanks for your answer.
I have made the following settings, but seems it doesn't work.
```
"retryStrategy": {
  "limit": "1",
  "retryPolicy": "OnTransientError"
},
```
```
{
  "name": "TRANSIENT_ERROR_PATTERN",
  "value": ".*Pod was active on the node longer than the specified deadline.*"
},
```

**Reply by terrytangyuan** [reply]
You should be able to modify your `activeDeadlineSeconds` in your template spec.

**Reply by zhengliquan** [reply]
I don't want to add time, what I need is retry. There is enough time now, it just gets stuck sometimes.

**Reply by tooptoop4** [reply]
@zhengliquan did u solve?
